internal/policies/mount/mount_test.go

```diff
@@ -0,0 +1,247 @@
+package mount_test
+
+import (
+	"context"
+	"os"
+	"os/user"
+	"path/filepath"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/adsys/internal/policies/entry"
+	"github.com/ubuntu/adsys/internal/policies/mount"
+	"github.com/ubuntu/adsys/internal/testutils"
+)
+
+func TestNew(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		readOnlyPerm bool
+
+		wantErr bool
+	}{
+		"creates manager successfully": {},
+
+		"creation fails with invalid runDir permissions": {readOnlyPerm: true, wantErr: true},
+	}
+
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			runDir := t.TempDir()
+			if tc.readOnlyPerm {
+				testutils.MakeReadOnly(t, runDir)
+			}
+
+			_, err := mount.New(mount.WithRunDir(runDir))
+			if tc.wantErr {
+				require.Error(t, err, "Expected an error when creating manager but got none.")
+				return
+			}
+			require.NoError(t, err, "Expected no error when creating manager but got one.")
+		})
+	}
+}
+
+func TestApplyPolicy(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		entries    []entry.Entry
+		key        string
+		isComputer bool
+
+		noEntries bool
+	}{
+		"successfully applies policy with valid key": {key: "user-mounts"},
+
+		"does nothing when trying to apply policy with user key and isComputer is set": {key: "user-mounts", isComputer: true},
+		"does nothing when trying to policy with unsupported key":                      {key: "not-supported-yet"},
+		"does nothing when trying to apply policy with no entries":                     {noEntries: true},
```

Comment:
I would put all users and (future computer) tests under the same umbrella here: users don’t care that this is internally 2 separate functions, but there is only one API call. So, having them all in the same table test makes more sense and make the API more discoverable.